recipes/ios-cmake/all/conanfile.py

```diff
@@ -1,26 +1,48 @@
-from conans import ConanFile, tools
-from conans.errors import ConanInvalidConfiguration
-
 import os
 
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.apple import is_apple_os
+from conan.tools.files import copy, get, save
+from conan.tools.layout import basic_layout
+
+required_conan_version = ">=1.52.0"
+
 
 class IosCMakeConan(ConanFile):
     name = "ios-cmake"
+    description = "iOS CMake toolchain to (cross) compile macOS/iOS/watchOS/tvOS"
     license = "BSD-3-Clause"
-    settings = "os" , "arch"
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://github.com/leetal/ios-cmake"
+    topics = ("apple", "ios", "cmake", "toolchain", "ios", "tvos", "watchos", "header-only")
+
+    package_type = "build-scripts"
+    settings = "os", "arch", "compiler", "build_type"
     options = {
         "enable_bitcode": [True, False],
         "enable_arc": [True, False],
         "enable_visibility": [True, False],
         "enable_strict_try_compile": [True, False],
-        "toolchain_target": ["auto", "OS", "OS64", "OS64COMBINED",
-                       "SIMULATOR", "SIMULATOR64", "SIMULATORARM64",
-                       "TVOS", "TVOSCOMBINED",
-                       "SIMULATOR_TVOS", "WATCHOS",
-                       "WATCHOSCOMBINED", "SIMULATOR_WATCHOS",
-                       "MAC", "MAC_ARM64", "MAC_CATALYST", "MAC_CATALYST_ARM64"]
+        "toolchain_target": [
+            "auto",
```

Comment:
I think we can use the fact that we're migrating this to Conan 2 to remove the auto value altogether so we don't need to check for it later? wdyt @valgur?

Reply:
I don't think that would work. The logic in `config_options()` sets the default value to an appropriate one based on `self.settings.os` and falls back to `auto` if it cannot be determined. The `self._default_toolchain_target` property does not cover several Apple OS-s, notably `Macos`.

Reply:
Oh so then the idea is that if it's not able to be guessed, we fail later in the validate if it's still set to auto? My only concern is that then running `conan create ... -o="&:toolchain_target=auto"` will always fail, the CLI value is overriding the decision in config_options

Reply:
Oh I see - I've pushed a minor change to let users pass auto and have it be still auto-guessed, let me know if it works for you :)

Reply:
The `build-scripts` package type is super-buggy or simply inconsistent with the options handling overall, for some reason. The user-supplied option values are simply ignored and not even validated to be one of the allowed values.

Reply:
Oooh... the options actually work, but only if I use `-o:b toolchain_target=...`.